pkg/ebpf/telemetry/errors_telemetry.go

```diff
@@ -0,0 +1,297 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package telemetry
+
+import (
+	"errors"
+	"fmt"
+	"hash"
+	"hash/fnv"
+	"sync"
+	"syscall"
+	"unsafe"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/asm"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/exp/slices"
+	"golang.org/x/sys/unix"
+
+	"github.com/DataDog/datadog-agent/pkg/network/ebpf/probes"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+	manager "github.com/DataDog/ebpf-manager"
+)
+
+const (
+	maxErrno    = 64
+	maxErrnoStr = "other"
+
+	ebpfMapTelemetryNS    = "ebpf_maps"
+	ebpfHelperTelemetryNS = "ebpf_helpers"
+)
+
+const (
+	readIndx int = iota
+	readUserIndx
+	readKernelIndx
+	skbLoadBytes
+	perfEventOutput
+)
+
+var ebpfMapOpsErrorsGauge = prometheus.NewDesc(fmt.Sprintf("%s__errors", ebpfMapTelemetryNS), "Failures of map operations for a specific ebpf map reported per error.", []string{"map_name", "error"}, nil)
+var ebpfHelperErrorsGauge = prometheus.NewDesc(fmt.Sprintf("%s__errors", ebpfHelperTelemetryNS), "Failures of bpf helper operations reported per helper per error for each probe.", []string{"helper", "probe_name", "error"}, nil)
+
+var helperNames = map[int]string{
+	readIndx:        "bpf_probe_read",
+	readUserIndx:    "bpf_probe_read_user",
+	readKernelIndx:  "bpf_probe_read_kernel",
+	skbLoadBytes:    "bpf_skb_load_bytes",
+	perfEventOutput: "bpf_perf_event_output",
+}
+
+// eBPFTelemetry struct contains all the maps that
+// are registered to have their telemetry collected.
+type eBPFTelemetry struct {
+	sync.Mutex
+	mapErrMap    *ebpf.Map
+	helperErrMap *ebpf.Map
+	mapKeys      map[string]uint64
+	probeKeys    map[string]uint64
+}
+
+// we are using one ebpfTelemetry object for all ebpf programs.
+var bpfTelemetry *eBPFTelemetry
+
+func init() {
+	bpfTelemetry = newEBPFTelemetry()
```

Comment:
Can we default to `nil` and call a constructor early on in `system-probe` lifetime? With the single agent binary, this would get created for everyone.